File: code/java/src/main/java/memento/Player.java
package memento;

class Player {
    private String state;

    public void setState(String state) {
        this.state = state;
    }

    public String getState() {
        return state;
    }

    public GameState saveStateToMemento() {
        return new GameState(state);
    }

    public void getStateFromMemento(GameState gameState) {
        state = gameState.getState();
    }
}

File: code/csharp/DesignPatterns/memento/Player.cs
namespace Memento;

class Player
{
    private string _state;

    public string State
    {
        get => _state;
        set => _state = value;
    }

    public GameState SaveStateToMemento()
    {
        return new GameState(_state);
    }

    public void GetStateFromMemento(GameState gameState)
    {
        _state = gameState.State;
    }
}
